src/game/objects/traps/movable_block.c

```diff
@@ -133,13 +139,18 @@ void MovableBlock_Collision(
 {
     ITEM_INFO *item = &g_Items[item_num];
 
+    if (Item_TestAnimEqual(lara_item, LA_STOP)) {
```

Comment:
This has a minor side effect when using the fly cheat. If you start pushing/pulling and then enter and exit fly mode, the block stops mid animation. Not really an issue for normal gameplay, but it's sometimes useful when flying quickly through a level e.g. in Natla's Mines at the drill. But it's really minor imo, not sure if it's worth handling. I don't believe it's possible otherwise for Lara's push/pull animation to be interrupted like this.

Reply:
We can do an or check for `g_Lara.water_status == LWS_CHEAT`.

This added variable is needed because Core reused item status `IS_ACTIVE` for when Lara moves a block which caused one of the bugs. `IS_ACTIVE` also is set when a triggered block falls. So we need to be able to differentiate.